pkg/store/proxy_heap.go

```diff
@@ -574,8 +574,9 @@ func newAsyncRespSet(
 		return nil, err
 	}
 
-	var labelsToRemove map[string]struct{}
-	if !st.SupportsWithoutReplicaLabels() && len(req.WithoutReplicaLabels) > 0 {
+	labelsToRemove := make(map[string]struct{})
+	dedupByInternalLabel := hasInternalReplicaLabels(st, req)
```

Comment:
Can we put this behind a feature flag? For Cortex, this code path seems unnecessary overhead

Reply:
Does cortex use the `proxy_heap`? But yes, we should add a FF in the bucket store

Reply:
Yeah proxy heap will be used by store gateway for lazy series? So it is always used.

Reply:
Ah yes, that's correct.

Reply:
Added a feature flag in https://github.com/fpetkovski/thanos/pull/5